scripts/prepare-subcharts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -e
+
+# Pull in the subcharts required by a chart and apply any patches on subcharts with fixed version numbers
```

Comment:
I think the flow and goal of this script is somewhat confusing. Can you add a high level overview of how it works. Something like: 

```
# In order to prepare sub-charts so that we can patch them: 
#  - We pull in any sub-chart that is a feature chart by running prepare on it
#  - We then call `helm dependency update` to pull in any other sub-chart
#  - And last we clean up any sub-chart that shouldn't have a patch applied to it.
```

Reply:
Modified this a little bit to add a more step-by-step overview:
```
# Pull in the subcharts required by a chart
#
# At the end of this process, all the required subcharts will exist in the charts/charts directory.
# Charts will either exist as chart archives (i.e. a .tgz file) if they are not eligible for patches or 
# as a local chart archive (i.e. a directory) if they are eligible for patches (i.e. remote subchart with fixed version number)
#
# In order to prepare sub-charts so that we can patch them: 
# - We apply any patches on the requirements.yaml to track any new subcharts added via patch or updates to version numbers of the subcharts
# - We pull in any subchart that is a feature chart by running prepare on it
# - We copy in any local subcharts that are located in the overlay/ directory
# - We call `helm dependency update` to generate chart archives (.tgz files) for local and remote charts
# - We clean up any copied charts from the overlay/ directory
# - We clean up any feature charts that were recursively prepared
# - We unarchive the chart archive if and only if its a remote subchart with a fixed version number (which means it can be patched)
# - We revert the initial patch we applied on the requirements.yaml
```

Let me know if you have any thoughts if this is too detailed or unclear!